corehq/apps/domain/forms.py

```diff
@@ -3122,10 +3124,53 @@ def __init__(self, *args, **kwargs):
             )
         )
 
-    # TODO: validate app_url
-    # 1. check if server is either production, india or EU
-    # 2. check if server is the same as the current server
-    # 3. check if app_url is a valid app url
+    def clean_app_url(self):
+        app_url = self.cleaned_data['app_url']
+
+        if not app_url.startswith('https://'):
+            raise forms.ValidationError(_("The URL must start with https://"))
+
+        server_mapping = {
+            'www': 'production',
+            'india': 'india',
+            'eu': 'eu'
+        }
+
+        is_valid_commcare_server = True
+        parsed_url = urllib.parse.urlparse(app_url)
+        netloc = parsed_url.netloc.split(".")
+        subdomain = netloc[0]
+
+        if len(netloc) == 3:
+            domain = netloc[1]
+            suffix = netloc[2]
+            if subdomain not in server_mapping.keys() or domain != 'commcarehq' or suffix != 'org':
+                is_valid_commcare_server = False
+        else:
+            is_valid_commcare_server = False
```

Comment:
could this be extracted into a staticmethod just to shorten up this whole `clean` function a little bit?

Reply:
e3f358ce5d1da09126b0bc3d73f0ff6493b15cc4